File: examples/python-repo/ecommerce/domain/order/order_line.py
from ecommerce.domain.order.money import Money


class OrderLine:
    def __init__(self, product_id, product_name, quantity, unit_price):
        if not product_name:
            raise ValueError("product name must not be empty")
        if quantity <= 0:
            raise ValueError("quantity must be positive")
        if unit_price is None:
            raise ValueError("unit price must not be None")

        self._product_id = product_id
        self._product_name = product_name
        self._quantity = quantity
        self._unit_price = unit_price

    @property
    def product_id(self):
        return self._product_id

    @property
    def product_name(self):
        return self._product_name

    @property
    def quantity(self):
        return self._quantity

    @property
    def unit_price(self):
        return self._unit_price

    def line_total(self):
        return self._unit_price.multiply(self._quantity)

File: examples/java-repo/src/main/java/com/example/ecommerce/domain/order/OrderLine.java
package com.example.ecommerce.domain.order;

public class OrderLine {

    private final long productId;
    private final String productName;
    private final int quantity;
    private final Money unitPrice;

    public OrderLine(long productId, String productName, int quantity, Money unitPrice) {
        if (productName == null || productName.isEmpty()) {
            throw new IllegalArgumentException("product name must not be empty");
        }
        if (quantity <= 0) {
            throw new IllegalArgumentException("quantity must be positive");
        }
        if (unitPrice == null) {
            throw new IllegalArgumentException("unit price must not be null");
        }
        this.productId = productId;
        this.productName = productName;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
    }

    public long getProductId() {
        return productId;
    }

    public String getProductName() {
        return productName;
    }

    public int getQuantity() {
        return quantity;
    }

    public Money getUnitPrice() {
        return unitPrice;
    }

    public Money lineTotal() {
        return unitPrice.multiply(quantity);
    }
}
